### PR TITLE
Improvements to HPE Aruba 2920 series

### PR DESCRIPTION
**Pull request by pv2b**
Some improvements to Aruba 2920 series:

- Incorrect part number was listed for HPE Aruba 2920-24G-PoE+
- Power figures were inconsistent (I've set allocated draw to idle and max draw to max theoretical, especially for PoE deployments these will need to be adjusted by end users according to their scenario)
- Added missing Aruba 2920-48G-PoE+-740W model
- Added weight information
- Added missing module bays for transceiver modules and stacking modules
- Added 3 new module types for this switch

Unfortunately airflow info is not in the datasheet.

### CODE REVIEW

File: device-types/HPE/Aruba-2920-48G-PoEP-740W.yaml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+part_number: J9836A
+u_height: 1
+is_full_depth: false
+weight: 5830
+weight_unit: g
+power-ports:
+  - name: PS1
+    type: iec-60320-c14
+    maximum_draw: 881
+    allocated_draw: 52
```
Comment (danner26) [inline device-types/HPE/Aruba-2920-48G-PoEP-740W.yaml:14]: It appears that the PSU is hot swappable. Thus it should be modelled under `module-bays`, and then the associated PSU module should be modelled in the module-types folder if they do not already exist there

If I am wrong about this please let me know and mark this as resolved

File: device-types/HPE/Aruba-2920-24G.yaml [hunk 2]
```diff
@@ -5,11 +5,13 @@ slug: hpe-aruba-2920-24g
...
 is_full_depth: false
+weight: 5250
+weight_unit: g
 power-ports:
   - name: PS1
     type: iec-60320-c14
-    maximum_draw: 165
-    allocated_draw: 58
+    maximum_draw: 58
+    allocated_draw: 26
```
Comment (danner26) [inline device-types/HPE/Aruba-2920-24G.yaml:14]: It appears that the PSU is hot swappable. Thus it should be modelled under `module-bays`, and then the associated PSU module should be modelled in the module-types folder if they do not already exist there

If I am wrong about this please let me know and mark this as resolved

File: device-types/HPE/Aruba-2920-48G-PoEP.yaml [hunk 3]
```diff
@@ -5,11 +5,13 @@ slug: hpe-aruba-2920-48g-poep
...
 is_full_depth: false
+weight: 5700
+weight_unit: g
 power-ports:
   - name: PS1
     type: iec-60320-c14
-    maximum_draw: 575
-    allocated_draw: 487
+    maximum_draw: 487
+    allocated_draw: 46
```
Comment (danner26) [inline device-types/HPE/Aruba-2920-48G-PoEP.yaml:14]: It appears that the PSU is hot swappable. Thus it should be modelled under `module-bays`, and then the associated PSU module should be modelled in the module-types folder if they do not already exist there

If I am wrong about this please let me know and mark this as resolved

File: device-types/HPE/Aruba-2920-48G.yaml [hunk 4]
```diff
@@ -5,11 +5,13 @@ slug: hpe-aruba-2920-48g
...
 is_full_depth: false
+weight: 5420
+weight_unit: g
 power-ports:
   - name: PS1
     type: iec-60320-c14
-    maximum_draw: 165
-    allocated_draw: 70
+    maximum_draw: 70
+    allocated_draw: 27
```
Comment (danner26) [inline device-types/HPE/Aruba-2920-48G.yaml:14]: It appears that the PSU is hot swappable. Thus it should be modelled under `module-bays`, and then the associated PSU module should be modelled in the module-types folder if they do not already exist there

If I am wrong about this please let me know and mark this as resolved

**Review by danner26 — CHANGES_REQUESTED** [review]

**Comment by danner26** [comment]
Please mark this PR as ready to review once you have corrected and resolved the requested changes

**Comment by pv2b** [comment]
Hello, regarding the hot-swappable power supplies, I initially reasoned that there was no use modelling them since there's only 1 PSU bay, and because of that there's no way to hot-swap them so modelling the supplies was kind of useless because there's no hotswap and only one valid configuration.

That said, I've re-examined the documentaiton and it seems the 2920 series has the ability to work with external power supplies using the XPS connector and a seperate external power supply shelf as per this documentation:

https://techhub.hpe.com/eginfolib/networking/docs/switches/WB/15-18/5998-8162_wb_2920_mcg/content/ch03s06.html

Probably what needs to happen to make this model complete is to add the HP 640 Redundant/External Power Supply Shelf (J9805A) as a device, and then model the XPS connector on the switches as as seperate Power Port, and then create the Power Supply Shelf device, with power outlets for each PSU?

**Comment by danner26** [comment]
It appears that all of the power inputs are modular, so they should be modelled as such. This goes for the dc power as well

**Comment by danner26** [comment]
You have this defined:
```
power-ports:
  - name: XPS
    type: dc-terminal
    maximum_draw: 475
    allocated_draw: 42
```

It appears all of the power is modular, so these should be modelled accordingly as well

**Comment by pv2b** [comment]
I have an idea what's got you confused. The "XPS" on the switch slot isn't a module bay, it's a port where you plug in an XPS cable, that does from the external redundant power supply shelf to the switch itself.

Please refer to page 2-10 of the (page 26 in th

> You have this defined:
> 
> ```
> power-ports:
>   - name: XPS
>     type: dc-terminal
>     maximum_draw: 475
>     allocated_draw: 42
> ```
> 
> It appears all of the power is modular, so these should be modelled accordingly as well

Oh!

I see where you're confused. No, this isn't a module slot, this is a ac.

You will deploy a HP 640 redundant power supply shelf, and then run XPS cables from that shelf to the XPS port on the switch.

Please Refer to the https://www.hpe.com/psnow/doc/c03952492

> You have this defined:
> 
> ```
> power-ports:
>   - name: XPS
>     type: dc-terminal
>     maximum_draw: 475
>     allocated_draw: 42
> ```
> 
> It appears all of the power is modular, so these should be modelled accordingly as well

Oh! I see where we're getting confused.

The switches can optionally take in external DC power from an Aruba 640 Redundant/External Power Supply Shelf. If you want redundant power, you'll stick one to three power supplies into one of these shelves. Then, you'll use a XPS cable to carry DC power down from those power supplies in the shelf to the XPS port on the switch itself.

The XPS power port represents this DC inlet on the switch itself.

You can refer to the [HP 640 RPS/EPS Installation and Power Setup Guide](https://www.hpe.com/psnow/doc/c03952492) for more information. On page 2-10 (page 26 in the PDF), see Figure 2-6, this will show you that the "XPS" on the switches is in fact a fixed DC input.

**Review by danner26 — APPROVED** [review]

**Pull request merged**